cmake/WoofSettings.cmake

```diff
@@ -74,6 +75,7 @@ if(MSVC)
 else()
     # We only want -Wall on GCC compilers, since /Wall on MSVC is noisy.
     _checked_add_compile_option(-Wall)
+    _checked_add_compile_option(-std=gnu23)
```

Comment:
Why is this required? We use union-of-function-pointers type quirks in the code which are valid in C99 but emit warnings with C23. The exact warnings that you override in your change in few lines above. Getting rid of these warnings would require a major work on the code that initializes the states and cheats arrays, or sticking to C99 for the time being.

Reply:
usleep and fileno only work with -std=gnu23 using clang and gcc

Reply:
> usleep and fileno only work with -std=gnu23 using clang and gcc

Why? We use usleep in `I_SleepUS` function, seems to work with C99: https://github.com/fabiangreffrath/woof/blob/8a736339a9bfd25032498147a55c656d03519a3f/src/i_timer.c#L226